### PR TITLE
#6233 fix sailor ignores error from maester; update test

### PR DESCRIPTION
**Pull request by Allirey**
[#6233](https://github.com/elasticio/elasticio/issues/6233)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +16,7 @@
     "node": ">=12.13.0"
   },
   "dependencies": {
-    "@elastic.io/object-storage-client": "1.0.1",
+    "@elastic.io/object-storage-client": "^2.0.0",
```
Comment (PaulAnnekov) [inline package.json:19]: https://www.notion.so/elasticio/Code-Style-89792419947546d2a52855d8df7a255b#0af15554a7d84cfcb52faaa9237281fa

File: lib/sailor.js [hunk 2]
```diff
@@ -300,6 +300,7 @@ class Sailor {
             id = await this.objectStorage.addAsStream(stream, this.settings.OBJECT_STORAGE_TOKEN);
         } catch (e) {
```
Comment (PaulAnnekov) [inline lib/sailor.js:301]: No need to catch it here at all. Will be caught by caller code.

File: lib/sailor.js [hunk 3]
```diff
@@ -386,38 +387,39 @@ class Sailor {
...
+                        try {
+                            [bodyId, ...passthroughIds] = await Promise.all([
+                                that.uploadMessageBody(bodyBuf),
+                                ...passthroughBufs.map(async ({ stepId, body, id }) => {
+                                    const bodyId = id || await that.uploadMessageBody(body);
+                                    return { stepId, bodyId };
+                                })
+                            ]);
+                        } catch (e) {
                             logger.info('Message body not uploaded');
```
Comment (PaulAnnekov) [inline lib/sailor.js:402]:
```suggestion
                            logger.error(e, 'Error during message/passthrough body upload');
```

File: lib/sailor.js [hunk 4]
```diff
@@ -386,38 +387,39 @@ class Sailor {
...
+                            e.message = `Maester error: ${e.toString()}`;
+                            return onError(e);
```
Comment (PaulAnnekov) [inline lib/sailor.js:404]:
```suggestion
                            return onError(new Error("Lightweight message/passthrough body upload error"));
```
This info will be enough for customer to see in errors. And additional details will be visible in logs.

**Review by PaulAnnekov — CHANGES_REQUESTED** [review]
You also need to update .circleci/config.yaml, to make it run `npm run audit` instead of `npm audit ...`.

File: package.json [hunk 5]
```diff
@@ -15,7 +16,7 @@
     "node": ">=12.13.0"
   },
   "dependencies": {
-    "@elastic.io/object-storage-client": "1.0.1",
+    "@elastic.io/object-storage-client": "2.0.0",
```
Comment (PaulAnnekov) [inline package.json:19]: version 2.0.0 has breaking change and test should fail in addAsStream. But 2.1.0 has reverted it. Use 2.1.0.

**Review by PaulAnnekov — CHANGES_REQUESTED** [review]

**Review by irisked — APPROVED** [review]

**Review by PaulAnnekov — APPROVED** [review]

**Pull request merged**